### PR TITLE
Implement flow read in v2 over nbworker

### PR DESCRIPTION
**Pull request by timofei-durakov**
Added:
- /v2/flows
- /v2/flows/{flow_id}
- /v2/flows{flow_id}/status

Changed:
- /v1/flows
- /v1/flows/{flow-id}
- /v1/flows/{flow-id}/status

v1 api started using nbworker topology for read requests as well

logic is moved to nb-worker topology

Related-With: #3291

### CODE REVIEW

File: src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/FlowOperationsBolt.java [hunk 1]
```diff
@@ -237,6 +243,25 @@ public void init() {
...
 
+    private List<FlowResponse> processFlowReadRequest(FlowReadRequest readRequest) {
+        try {
+            Flow f = flowOperationsService.getFlow(readRequest.getFlowId());
+            FlowDto dto = FlowMapper.INSTANCE.map(f);
+            FlowResponse response = new FlowResponse(dto);
+            return Collections.singletonList(response);
+        } catch (FlowNotFoundException e) {
+            throw new MessageException(ErrorType.NOT_FOUND, e.getMessage(),
+                    "Could not get connected devices for non existent flow");
```
Comment (niksv) [inline src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/FlowOperationsBolt.java]: I think error message is wrong.
 `connected devices`

File: src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowControllerV2.java [hunk 2]
```diff
@@ -78,6 +81,31 @@
...
+    /**
+     * Gets flow.
+     *
+     * @param flowId        flow id
+     * @return flow
+     */
+    @ApiOperation(value = "Gets flow", response = FlowResponsePayload.class)
+    @GetMapping(value = "/{flow-id:.+}")
+    @ResponseStatus(HttpStatus.OK)
+    public CompletableFuture<FlowResponseV2> getFlow(@PathVariable(name = "flow-id") String flowId) {
```
Comment (niksv) [inline src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowControllerV2.java]: `flow-id` please use `flow_id` like in other V2 API

**Comment by dpoltavets** [comment]
Need to add `/v2/flows/status/{flow-id}`

**Comment by timofei-durakov** [comment]
@dpoltavets added status to v2 as well

File: src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowControllerV2.java [hunk 3]
```diff
@@ -78,6 +82,43 @@
         return flowService.deleteFlowV2(flowId);
     }
 
+    /**
+     * Gets flow.
+     *
+     * @param flowId        flow id
+     * @return flow
+     */
+    @ApiOperation(value = "Gets flow", response = FlowResponsePayload.class)
```
Comment (rtretyak) [inline src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowControllerV2.java]: FlowResponseV2

File: src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowControllerV2.java [hunk 4]
```diff
@@ -78,6 +82,43 @@
...
+    public CompletableFuture<FlowResponseV2> getFlow(@PathVariable(name = "flow_id") String flowId) {
+        return flowService.getFlowV2(flowId);
+    }
+
+    /**
+     * Dumps all flows. Dumps all flows with specific status if specified.
+     *
+     * @return list of flow
+     */
+    @ApiOperation(value = "Dumps all flows", response = FlowResponsePayload.class, responseContainer = "List")
```
Comment (rtretyak) [inline src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowControllerV2.java]: FlowResponseV2

File: src-java/nbworker-topology/nbworker-messaging/src/main/java/org/openkilda/messaging/nbtopology/request/FlowReadRequest.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Represents flow read northbound request.
+ */
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class FlowReadRequest extends FlowsBaseRequest {
+
+    private static final long serialVersionUID = 1L;
+
+    @JsonProperty("flowid")
```
Comment (sergii-iakovenko) [inline src-java/nbworker-topology/nbworker-messaging/src/main/java/org/openkilda/messaging/nbtopology/request/FlowReadRequest.java]: nit: as per the convention, it should be "flow_id"

File: src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/FlowMapper.java [hunk 6]
```diff
@@ -215,9 +219,13 @@ default String encodeFlowState(FlowState state) {
         return state.getState();
     }
 
+    @Mapping(target = "mainPath", source = "mainFlowPathStatus")
+    @Mapping(target = "protectedPath", source = "protectedFlowPathStatus")
+    PathStatus map(FlowStatusDetails flowStatusDetails);
```
Comment (sergii-iakovenko) [inline src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/FlowMapper.java]: nit: does it make sense to have a dedicated mapper for PathStatus?

**Pull request merged**